src/typescript.coffee

```diff
@@ -0,0 +1,104 @@
+###
+Cache for source code transpiled by TypeScript.
+
+Inspired by https://github.com/atom/atom/blob/7a719d585db96ff7d2977db9067e1d9d4d0adf1a/src/babel.coffee
+###
+
+crypto = require 'crypto'
+fs = require 'fs-plus'
+path = require 'path'
+tss = null # Defer until used
+
+stats =
+  hits: 0
+  misses: 0
+
+defaultOptions =
+  target: 1 # ES5
```

Comment:
Yes. Typescript supports a superset of ES6 futures supported by IO.js

If Typescript spits out ES6 it might not run natively.